### DISCUSSION TITLE
Official database schema? (MSSQL)

### DISCUSSION THREAD

**Discussion by aukevanleeuwen** [kagkarlsson/db-scheduler#453] [Q&A] [answered]
Hello,

Ik want to use this library in a Spring Boot (3) application. To evolve the database schema I would like the 'official' schema for my database Microsoft SQL server (Azure). 

I find a reference here in the test code: https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/test/resources/mssql_tables.sql

But I'm wondering if this is the official one? Is it?

Also: if it is the official one: I see this column:

```
  picked_by text,
```

However I get this warning:
>   Since SQL Server 2016 the 'text' type is deprecated. Use 'varchar(max)' instead 

Also: looking at other database schemas (h2, mysql) I see that `varchar(50)` is used. Should this be used instead? Is that guaranteed to be at most 50 chars?

**Comment by kagkarlsson** [comment, marked as answer]
These are meant to be the official schemas yeah, but they have all been contributed by different users. (except for postgres)

If there is a deprecation warning, feel free to report an issue for updating the schema.

The name is truncated to 50 Java characters:

```java
String pickedBy = truncate(schedulerSchedulerName.getName(), 50);
```

**Comment by aukevanleeuwen** [comment]
Created https://github.com/kagkarlsson/db-scheduler/pull/455 for it.
